libs/ui/src/pure/DismissableInlineBanner/index.tsx

```
@@ -8,7 +8,7 @@ interface DismissableInlineBannerProps extends Omit<InlineBannerProps, 'onClose'
 }
 
 export function DismissableInlineBanner(props: DismissableInlineBannerProps): ReactNode {
-  return ClosableBanner(props.bannerId, (onClose) => {
-    return <InlineBanner {...props} onClose={onClose} />
-  })
+  return (
+    <ClosableBanner storageKey={props.bannerId} callback={(onClose) => <InlineBanner {...props} onClose={onClose} />} />
```

ESLint

[error] Do not define components during render. React will see a new component type on every render and destroy the entire subtree’s DOM nodes and state (https://reactjs.org/docs/reconciliation.html#elements-of-different-types). Instead, move this component definition out of the parent component “DismissableInlineBanner” and pass data as props. If you want to allow component creation in props, set allowAsProps option to true.

(react/no-unstable-nested-components)